packages/block-editor/src/hooks/effects.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * WordPress dependencies
+ */
+import { hasBlockSupport } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import StylesEffectsPanel, {
+	useHasEffectsPanel,
+} from '../components/global-styles/effects-panel';
+import { InspectorControls } from '../components';
+import { store as blockEditorStore } from '../store';
+
+export const EFFECTS_SUPPORT_KEYS = [ 'shadow' ];
+
+export function hasEffectsSupport( blockName ) {
+	return EFFECTS_SUPPORT_KEYS.some( ( key ) =>
+		hasBlockSupport( blockName, key )
+	);
+}
+
+function EffectsInspectorControl( { children, resetAllFilter } ) {
+	return (
+		<InspectorControls group="effects" resetAllFilter={ resetAllFilter }>
+			{ children }
+		</InspectorControls>
+	);
+}
+export function EffectsPanel( { clientId, setAttributes, settings } ) {
+	const isEnabled = useHasEffectsPanel( settings );
+	const blockAttributes = useSelect(
+		( select ) => select( blockEditorStore ).getBlockAttributes( clientId ),
+		[ clientId ]
+	);
+	const shadow = blockAttributes?.style?.shadow;
+	const value = useMemo(
+		() => ( {
+			shadow,
+		} ),
+		[ shadow ]
+	);
```

Comment:
Unsure about this; I kept `useMemo`, because we were calculating the value in a function, but now we're just accessing the value directly; we might want to do the following instead:

```suggestion
	const value = { shadow };
```

Thoughts?

Reply:
Yes, `shadow` variable can be used instead.